src/System.Threading.AccessControl/tests/ThreadingAclExtensionsTests.cs

```diff
@@ -45,5 +54,161 @@ public static void ExistenceTest_Unix()
+            });
+        }
+
+        [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData((EventResetMode)int.MinValue)]
+        [InlineData((EventResetMode)(-1))]
+        [InlineData((EventResetMode)2)]
+        [InlineData((EventResetMode)int.MaxValue)]
+        public static void EventWaitHandle_Create_InvalidMode(EventResetMode mode)
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("mode", () =>
+            {
+                using EventWaitHandle eventHandle = EventWaitHandleAcl.Create(initialState: true, mode, "name", out bool createdNew, GetBasicEventWaitHandleSecurity());
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void EventWaitHandle_Create_BasicSecurity()
+        {
+            var security = GetBasicEventWaitHandleSecurity();
+            VerifyEventWaitHandle(security);
+        }
+
+        [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData(true,  EventResetMode.AutoReset,   AccessControlType.Allow, Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(false, EventResetMode.AutoReset,   AccessControlType.Allow, Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(true,  EventResetMode.ManualReset, AccessControlType.Allow, Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(false, EventResetMode.ManualReset, AccessControlType.Allow, Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(true,  EventResetMode.AutoReset,   AccessControlType.Deny,  Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(false, EventResetMode.AutoReset,   AccessControlType.Deny,  Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(true,  EventResetMode.ManualReset, AccessControlType.Deny,  Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        [InlineData(false, EventResetMode.ManualReset, AccessControlType.Deny,  Interop.Kernel32.MAXIMUM_ALLOWED | Interop.Kernel32.SYNCHRONIZE | Interop.Kernel32.EVENT_MODIFY_STATE)]
+        public static void EventWaitHandle_Create_SpecificParameters(bool initialState, EventResetMode mode, AccessControlType accessControl, int rights)
+        {
+            var security = GetEventWaitHandleSecurity(WellKnownSidType.BuiltinUsersSid, (EventWaitHandleRights)rights, accessControl);
+            if (accessControl == AccessControlType.Deny)
+            {
+                Assert.Throws<UnauthorizedAccessException>(() =>
```

Comment:
@JeremyKuhne I don't think this is the expected behavior, but I added the Assert.Throws to illustrate what I'm seeing:
- In netcore, we throw UnauthorizedAccessException when AccessControlType is deny.
- In netfx, we don't throw anything.
I'm comparing the code to try to understand why the behavior doesn't match different frameworks.